includes/User.php

```diff
@@ -4841,7 +4841,29 @@ protected function saveOptions() {
 			}
 		}
 
-		$dbw->delete( 'user_properties', array( 'up_user' => $this->getId() ), __METHOD__ );
+		// kinda ghetto, but :(
+		if ($wgPreferencesUseService) {
+			$regex = implode ('|', $wgUserPreferenceWhiteList['regexes']);
```

Comment:
Right. I forgot that it will be handled in the same request so there will be no race conditions (which could be problematic when saving user preference in different request/thread). Am I right? There will be no case when we try to save in thread user_pref1 and it will not be returned by getPreferences() in other thread, but it will be saved before delete on 'user_preference' and thus lost at the end?

```getPreferences() [thread1] -> savePreferences('user-pref1') [thread2] -> delete('user_preferences') [thread2]```